src/displayapp/screens/Clock.cpp

```diff
@@ -10,243 +10,73 @@
-  lv_label_set_text(bleIcon, Symbols::bluetooth);
-  lv_obj_align(bleIcon, batteryPlug, LV_ALIGN_OUT_LEFT_MID, -5, 0);
-
-  notificationIcon = lv_label_create(lv_scr_act(), NULL);
-  lv_label_set_text(notificationIcon, NotificationIcon::GetIcon(false));
-  lv_obj_align(notificationIcon, nullptr, LV_ALIGN_IN_TOP_LEFT, 10, 0);
-
-  label_date = lv_label_create(lv_scr_act(), nullptr);
-
-  lv_obj_align(label_date, lv_scr_act(), LV_ALIGN_IN_LEFT_MID, 0, 60);
-
-  label_time = lv_label_create(lv_scr_act(), nullptr);
-
-  lv_obj_set_style_local_text_font(label_time, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, &jetbrains_mono_extrabold_compressed);
-
-  lv_obj_align(label_time, lv_scr_act(), LV_ALIGN_IN_LEFT_MID, 0, 0);
-
-  backgroundLabel = lv_label_create(lv_scr_act(), nullptr);
-  backgroundLabel->user_data = this;
-  lv_obj_set_click(backgroundLabel, true);
-  lv_obj_set_event_cb(backgroundLabel, event_handler);
-  lv_label_set_long_mode(backgroundLabel, LV_LABEL_LONG_CROP);
-  lv_obj_set_size(backgroundLabel, 240, 240);
-  lv_obj_set_pos(backgroundLabel, 0, 0);
-  lv_label_set_text(backgroundLabel, "");
-
-
-  heartbeatIcon = lv_label_create(lv_scr_act(), nullptr);
-  lv_label_set_text(heartbeatIcon, Symbols::heartBeat);
-  lv_obj_align(heartbeatIcon, lv_scr_act(), LV_ALIGN_IN_BOTTOM_LEFT, 5, -2);
-
-  heartbeatValue = lv_label_create(lv_scr_act(), nullptr);
-  lv_label_set_text(heartbeatValue, "0");
-  lv_obj_align(heartbeatValue, heartbeatIcon, LV_ALIGN_OUT_RIGHT_MID, 5, 0);
-
-  heartbeatBpm = lv_label_create(lv_scr_act(), nullptr);
-  lv_label_set_text(heartbeatBpm, "BPM");
-  lv_obj_align(heartbeatBpm, heartbeatValue, LV_ALIGN_OUT_RIGHT_MID, 5, 0);
-
-  stepValue = lv_label_create(lv_scr_act(), nullptr);
-  lv_label_set_text(stepValue, "0");
-  lv_obj_align(stepValue, lv_scr_act(), LV_ALIGN_IN_BOTTOM_RIGHT, -5, -2);
-
-  stepIcon = lv_label_create(lv_scr_act(), nullptr);
-  lv_label_set_text(stepIcon, Symbols::shoe);
-  lv_obj_align(stepIcon, stepValue, LV_ALIGN_OUT_LEFT_MID, -5, 0);
-}
+        Controllers::Settings &settingsController,
+        Controllers::HeartRateController& heartRateController) : Screen(app),
+        dateTimeController{dateTimeController}, batteryController{batteryController},
+        bleController{bleController}, notificatioManager{notificatioManager},
+        settingsController{settingsController},
+        heartRateController{heartRateController},
+        screens{app, 
+          settingsController.GetClockFace(),
+          {
+                [this]() -> std::unique_ptr<Screen> { return WatchFaceDigitalScreen(); },
+                [this]() -> std::unique_ptr<Screen> { return WatchFaceAnalogScreen(); },
+                //[this]() -> std::unique_ptr<Screen> { return WatchFaceMinimalScreen(); },
+                //[this]() -> std::unique_ptr<Screen> { return WatchFaceCustomScreen(); }
+          },
+          Screens::ScreenListModes::LongPress          
+        } {
+
+          settingsController.SetAppMenu(0);
```

Comment:
Is this necessary? I think the app menu index is handled by Tile ?

Reply:
One of the changes made in this PR is to save the position of the application menu for when leaving an app we return to the menu where it is and not to the beginning.
Here we reset this value to go back to the first page when we call the laucher from the clock.